client.go

```diff
@@ -456,16 +456,16 @@ func (d *dataCloser) Close() error {
 			expectedResponses--
 		}
 		return nil
-	} else {
-		_, _, err := d.c.Text.ReadResponse(250)
-		if err != nil {
-			if protoErr, ok := err.(*textproto.Error); ok {
-				return toSMTPErr(protoErr)
-			}
-			return err
+	}
+
+	_, _, err := d.c.Text.ReadResponse(250)
+	if err != nil {
+		if protoErr, ok := err.(*textproto.Error); ok {
+			return toSMTPErr(protoErr)
 		}
-		return nil
+		return err
 	}
+	return nil
```

Comment:
Hm, why this change in `dataCloser.Close`? I don't think it's correct, I'd assume it breaks LMTP.

Reply:
Ah actually I meant to ask you about this. gopls was complaining about the block so I adjusted it and ran tests (passed) so I assumed it was fine. Was going to ask first and forgot. I'll revert this though.

Reply:
Oh, I missed the `return` in the `if` branch above. So I guess both are the same.

I'd prefer to keep it the way it was before though, because it makes it clearer what the two cases are.